### PR TITLE
user docs: Adds network gateway use-cases.

### PR DESCRIPTION
**Pull request by tobsan**
Also splits the gateway documentation source files into one for each
gateway. They are all included from the index file.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

**Review by kursatkobya — APPROVED** [review]

File: doc/chapters/gateways/10-network-gateway.rst [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+``host`` can be specific hostname or ip address and also ``*`` indicating all available ip sources.
+
+When ``ports`` is not specified, the rule applies to all ports.  ``ports`` are valid between 0 and
+65536. Ports can be specified as a single port, as a string with a range ``"X-Y"``, or as a list
+``[X,Y,Z]``.
+
+There are three valid values for ``protocols``: ``"tcp"``, ``"udp"`` and ``"icmp"``. When
+``protocols`` is not specified in the ``allow`` list item, the rule applies to all protocols. When
+there is an item which has ports without protocols, tcp protocol will be applied to the rule.
```
Comment (kursatkobya) [inline doc/chapters/gateways/10-network-gateway.rst]: can you change this line like
"When there is an item which has ports without protocols, the rule applies to all protocols."

Comment (tobsan) [reply doc/chapters/gateways/10-network-gateway.rst]: Sure.

**Review by DunderRoffe — CHANGES_REQUESTED** [review]
Please choose a more relevant name for the commit. Changing the structure of the docs is a bit big to not mention in the commit name.

**Comment by tobsan** [comment]
I've changed the name of the commit now.

**Review by DunderRoffe — APPROVED** [review]

**Pull request merged**